### PR TITLE
fix `Avoid expanding increment operator for complex left hand values` (close #2029)

### LINKED ISSUE #2029: Avoid expanding increment operator for complex left hand values
Page http://simionov-10:20020/playground/react/index.html

The problematic expression is
 ```
(e = this._group.getBBox()).height += i.top + i.bottom - this._titleTextY;
```

Hammerhead processes it to 
```
 (e = this._group.getBBox()).height =(e = this._group.getBBox()).height + i.top + i.bottom - this._titleTextY;
```

The `(e = this._group.getBBox())` expression can have side effects, and the result of the proccessed expression can differ to the result of the original one. We should avoid expanding `+=` operator in such cases.

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest-all

File: test/server/script-processor-test.js [hunk 1]
```diff
@@ -320,10 +320,12 @@ describe('Script processor', () => {
...
 
-    it('Should expand concat operator', () => {
+    it('Should expand concat operator only if necessary', () => {
         testProcessing([
-            { src: 'prop += 1', expected: 'prop = prop + 1' },
-            { src: 'prop += 2 + prop + 1', expected: 'prop = prop + (2 + prop + 1)' }
+            { src: 'prop += 1', expected: 'prop += 1' },
+            { src: 'prop += 2 + prop + 1', expected: 'prop += 2 + prop + 1' },
+            { src: 'prop.href += 1', expected: '__set$(prop,"href",__get$(prop,"href")+1)' },
+            { src: 'prop.location += 2 + prop.location + 1', expected: '__set$(prop,"location",__get$(prop,"location")+(2 + __get$(prop,"location") + 1 ))' }
```
Comment (AndreyBelym) [inline test/server/script-processor-test.js]: I would like to see the original `(a = b) += c` scenario here.

File: test/server/script-processor-test.js [hunk 2]
```diff
@@ -320,10 +320,12 @@ describe('Script processor', () => {
         ]);
     });
 
-    it('Should expand concat operator', () => {
+    it('Should expand concat operator only if necessary', () => {
         testProcessing([
-            { src: 'prop += 1', expected: 'prop = prop + 1' },
-            { src: 'prop += 2 + prop + 1', expected: 'prop = prop + (2 + prop + 1)' }
+            { src: 'prop += 1', expected: 'prop += 1' },
+            { src: 'prop += 2 + prop + 1', expected: 'prop += 2 + prop + 1' },
```
Comment (AndreyBelym) [inline test/server/script-processor-test.js]: Why this expression is not instrumented? I thought that the problem was caused by side-effects in the **left-hand** side of the assignment. The right-hand part can be as complex as you wish.

Comment (AndreyBelym) [reply test/server/script-processor-test.js]: Otherwise, we won't handle cases like `location += '/subpath'`

**Pull request merged**